Panic in `core.initializeDEXConnections`

It seems `Core` is trying to access a method on a non-existent `*xcWallet`.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x100ca15f7]

goroutine 1261 [running]:
decred.org/dcrdex/client/core.(*xcWallet).state(0x0)
        /Users/philemon/go/pkg/mod/github.com/ukane-philemon/dcrdex@v0.0.0-20240107001126-c615fb9e6572/client/core/wallet.go:270 +0x17
decred.org/dcrdex/client/core.(*Core).initializeDEXConnections(0xc003e30000, {0x101debb20, 0xc006b4e000})
        /Users/philemon/go/pkg/mod/github.com/ukane-philemon/dcrdex@v0.0.0-20240107001126-c615fb9e6572/client/core/core.go:5235 +0x489
decred.org/dcrdex/client/core.(*Core).Login(0xc003e30000, {0xc0062989b0, 0x1, 0x8})
        /Users/philemon/go/pkg/mod/github.com/ukane-philemon/dcrdex@v0.0.0-20240107001126-c615fb9e6572/client/core/core.go:4746 +0x6aa
github.com/crypto-power/cryptopower/dexc.(*DEXClient).Login(0xc003d8efc0, {0xc0062989b0?, 0x1?, 0xc000000000?})
        /Users/philemon/Desktop/osp/cryptopower/dexc/core.go:56 +0x25

```

--- a/client/core/core.go
+++ b/client/core/core.go
@@ -5231,10 +5231,13 @@ func (c *Core) initializeDEXConnections(crypter encrypt.Crypter) {
 				unbip(bondAssetID), targetTier, dc.acct.host, maxBondedAmt)
 			wallet, exists := c.wallet(bondAssetID)
 			if !exists || !wallet.connected() { // connectWallets already run, just fail
-				subject, _ := c.formatDetails(TopicWalletConnectionWarning)
-				c.notify(newWalletConfigNote(TopicWalletConnectionWarning, subject,
-					fmt.Sprintf("bond asset wallet %s not configured or connected", unbip(bondAssetID)),
-					db.ErrorLevel, wallet.state()))
+				details := fmt.Sprintf("bond asset wallet %s not configured or connected", unbip(bondAssetID))
+				if exists {
+					subject, _ := c.formatDetails(TopicWalletConnectionWarning)
+					c.notify(newWalletConfigNote(TopicWalletConnectionWarning, subject, details, db.ErrorLevel, wallet.state()))
+				} else {
+					c.log.Debug(details)
+				}
 			} else if !wallet.unlocked() {
 				err = wallet.Unlock(crypter)
 				if err != nil {

--- a/client/core/core.go
+++ b/client/core/core.go
@@ -5231,13 +5231,12 @@ func (c *Core) initializeDEXConnections(crypter encrypt.Crypter) {
 				unbip(bondAssetID), targetTier, dc.acct.host, maxBondedAmt)
 			wallet, exists := c.wallet(bondAssetID)
 			if !exists || !wallet.connected() { // connectWallets already run, just fail
-				details := fmt.Sprintf("bond asset wallet %s not configured or connected", unbip(bondAssetID))
+				subject, details := c.formatDetails(TopicBondWalletNotConnected, unbip(bondAssetID))
+				var w *WalletState
 				if exists {
-					subject, _ := c.formatDetails(TopicWalletConnectionWarning)
-					c.notify(newWalletConfigNote(TopicWalletConnectionWarning, subject, details, db.ErrorLevel, wallet.state()))
-				} else {
-					c.log.Debug(details)
+					w = wallet.state()
 				}
+				c.notify(newWalletConfigNote(TopicBondWalletNotConnected, subject, details, db.ErrorLevel, w))
 			} else if !wallet.unlocked() {
 				err = wallet.Unlock(crypter)
 				if err != nil {

--- a/client/core/locale_ntfn.go
+++ b/client/core/locale_ntfn.go
@@ -57,6 +57,10 @@ var originLocale = map[Topic]*translation{
 		subject:  "Wallet connection warning",
 		template: "Incomplete registration detected for %s, but failed to connect to the Decred wallet",
 	},
+	TopicBondWalletNotConnected: {
+		subject:  "Bond wallet not connected",
+		template: "Wallet for selected bond asset %s is not connected",
+	},
 	// [host, error]
 	TopicWalletUnlockError: {
 		subject:  "Wallet unlock error",

--- a/client/core/notification.go
+++ b/client/core/notification.go
@@ -582,6 +582,7 @@ const (
 	TopicWalletPeersWarning         Topic = "WalletPeersWarning"
 	TopicWalletTypeDeprecated       Topic = "WalletTypeDeprecated"
 	TopicWalletPeersUpdate          Topic = "WalletPeersUpdate"
+	TopicBondWalletNotConnected     Topic = "BondWalletNotConnected"
 )
 
 func newWalletConfigNote(topic Topic, subject, details string, severity db.Severity, walletState *WalletState) *WalletConfigNote {

--- a/client/webserver/site/src/js/app.ts
+++ b/client/webserver/site/src/js/app.ts
@@ -657,7 +657,8 @@ export default class Application {
       case 'walletconfig': {
         // assets can be null if failed to connect to dex server.
         if (!assets) return
-        const wallet = (note as WalletConfigNote).wallet
+        const wallet = (note as WalletConfigNote)?.wallet
+        if (!wallet) return
         const asset = assets[wallet.assetID]
         asset.wallet = wallet
         walletMap[wallet.assetID] = wallet

